client/src/extension.ts

```diff
@@ -6,7 +6,7 @@
 import * as vscode from 'vscode'
 import { type LanguageClient } from 'vscode-languageclient/node'
 
-import { ClientNotificationManager } from './ui/ClientNotificationManager'
+import { MessageManager } from './ui/MessageManager'

```

Comment:
Could you detail the reason for this renaming? I think notification was the right term since it popped notifications on the UI.

Reply:
Well, that's confusing, and I realize my description of the issue was not accurate.

In fact, the VS Code's doc also uses the therm "notification" for these: https://code.visualstudio.com/api/ux-guidelines/notifications. However, the api uses the term "message" for the popups showed to the user: [showInformationMessage](https://code.visualstudio.com/api/references/vscode-api#window.showInformationMessage), [showWarningMessage](https://code.visualstudio.com/api/references/vscode-api#window.showWarningMessage), [showErrorMessage](https://code.visualstudio.com/api/references/vscode-api#window.showErrorMessage).

Meanwhile, the language server uses the term "notification" for the information sent from the client to the server, or from the server to the client: [sendNotification](https://github.com/microsoft/vscode-languageserver-node/blob/277e4564193c4ed258fe4d8d405d3379665bbab9/protocol/src/common/connection.ts#L101), [onNotification](https://github.com/microsoft/vscode-languageserver-node/blob/277e4564193c4ed258fe4d8d405d3379665bbab9/protocol/src/common/connection.ts#L141).

I am trying to solve this confusion. Maybe "PopupManager" would be even better ?

Reply:
Yes Popup is perfectly clear.

Reply:
The popup is because of the `window.showInformationMessage` and other window APIs you mentioned above. If this class is only responsible for sending or receiving notifications, I don't think the rename is needed. Because you can do more than just calling `window.showInformationMesssage` in the `onNotification` callback.

Reply:
You can also show popups that do not require calls to onNotification, and use it for stuff that won't trigger a popup. For example, sendNotification is called in languageClient in order to notify the server that a file has changed. This use case is totally unrelated to to the purpose of PopupManager, and I'd argue keeping the name NotificationManager for it is confusing.

Reply:
It really depends on what you use this class for or you can get rid of it since there isn't a lot of stuff.

Reply:
I think there is confusion from this class because it does many different things:

1. Act as a GUI popup provider
2. Manage language server notifications
3. List and handle error types

Those things should be decoupled into multiple files. Example:

1. client/gui/notificationsPopup.ts
2. client/language/notificationsHandler.ts
3. client/lib/bitbakeErrors.ts

The same could be done on the server side.